src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -42,9 +40,12 @@ public async ValueTask<ITimeZone> GetUserTimeZoneAsync()
         return _clock.GetTimeZone(currentTimeZoneId);
     }
 
-    public async ValueTask UpdateUserTimeZoneAsync(IUser user)
+    /// <inheritdoc/>
+    public async ValueTask UpdateAsync(IUser user)
     {
-        var userName = user?.UserName;
+        ArgumentNullException.ThrowIfNull(user);
+
+        var userName = user.UserName;
```

Comment:
Why the extra allocation?

Reply:
Not too much, I used instead of using `user.userName` multiple times

Reply:
But you can just access the user.UserName without the need to extra allocation